packages/backend/server/src/core/mail/job.ts

```diff
@@ -0,0 +1,140 @@
+import { Injectable } from '@nestjs/common';
+import { getStreamAsBuffer } from 'get-stream';
+
+import { JOB_SIGNAL, OnJob } from '../../base';
+import { type MailName, MailProps, Renderers } from '../../mails';
+import { UserProps, WorkspaceProps } from '../../mails/components';
+import { Models } from '../../models';
+import { DocReader } from '../doc/reader';
+import { WorkspaceBlobStorage } from '../storage';
+import { MailSender, SendOptions } from './sender';
+
+type DynamicallyFetchedProps<Props> = {
+  [Key in keyof Props]: Props[Key] extends infer Prop
+    ? Prop extends UserProps
+      ? {
+          $$userId: string;
+        } & Omit<Prop, 'email' | 'avatar'>
+      : Prop extends WorkspaceProps
+        ? {
+            $$workspaceId: string;
+          } & Omit<Prop, 'name' | 'avatar'>
+        : Prop
+    : never;
+};
+
+type SendMailJob<Mail extends MailName = MailName, Props = MailProps<Mail>> = {
+  name: Mail;
+  to: string;
+  // NOTE(@forehalo):
+  //   workspace avatar currently send as base64 img instead of a avatar url,
+  //   so the content might be too large to be put in job payload.
```

Comment:
https://app.graphite.dev/github/pr/toeverything/AFFiNE/10964/refactor(server)-improve-workspace-content-retrieval-from-database?ref=gt-system-tray-app let use avatarUrl instead of base64 image

Reply:
the blob api is protected behind access controller, it's can't be used as public link